databroker/core.py

```diff
@@ -607,6 +608,31 @@ def stream_gen(entry):
     yield ('stop', stop)
 
 
+class ArgsDict(dict):
+    """
+    This class is made so that we can define our own tokenize method, which
+    greatly speeds up the time to load a BlueskyRun.
+    """
+    ...
+
+# Register a tokenize method for ArgsDict
+@normalize_token.register(ArgsDict)
+def tokenize_dict(args):
+    return (descriptor['uid'] for descriptor in args['metadata']['descriptors'])
+
+class MetaDict(dict):
+    """
+    This class is made so that we can define our own tokenize method, which
+    greatly speeds up the time to load a BlueskyRun.
+    """
+    ...
+
+# Register a tokenize method for MetaDict
+@normalize_token.register(MetaDict)
+def tokenize_dict(meta):
+    return (descriptor['uid'] for descriptor in meta['descriptors'])
```

Comment:
As written, this returns a generator instance. A tuple is probably intended.
```suggestion
    return (descriptor['uid'] for descriptor in meta['descriptors'])
```